api/client/commands.go

```diff
@@ -162,21 +163,19 @@ func (cli *DockerCli) CmdBuild(args ...string) error {
 			}
 
 			// Verify that 'filename' is within the build context
-			if !strings.HasSuffix(absRoot, string(os.PathSeparator)) {
-				absRoot += string(os.PathSeparator)
-			}
-			if !strings.HasPrefix(filename, absRoot) {
-				return fmt.Errorf("The Dockerfile (%s) must be within the build context (%s)", *dockerfileName, root)
+			filename, err = symlink.FollowSymlinkInScope(filename, absRoot)
```

Comment:
@icecrime That's not something we want. Ironically, I fixed that in an earlier version of the code, but introduced another bug. Fixing this now thanks!